qcengine/programs/adcc.py

```diff
@@ -0,0 +1,125 @@
+        Parameters
+        ----------
+        raise_error: bool
+            Passed on to control negative return between False and ModuleNotFoundError raised.
+        Returns
+        -------
+        bool
+            If adcc and psi4 are found, returns True.
+            If raise_error is False and adcc or psi4 is missing, returns False.
+            If raise_error is True and adcc or psi4 are missing, the error message is raised.
+        """
+        found_adcc = which_import(
+            "adcc",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install via `conda install adcc -c adcc`.",
+        )
+        found_psi4 = which_import(
+            "psi4",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install psi4 for adcc harness via `conda install psi4 -c psi4`.",
+        )
+        return found_adcc and found_psi4
+
+    def get_version(self) -> str:
+        """Return the currently used version of adcc"""
+        self.found(raise_error=True)
+
+        which_prog = which_import("adcc")
+        if which_prog not in self.version_cache:
+            import adcc
+
+            self.version_cache[which_prog] = safe_version(adcc.__version__)
+        return self.version_cache[which_prog]
+
+    def compute(self, input_model: "AtomicInput", config: "TaskConfig") -> "AtomicResult":
+        """
+        Runs adcc
+        """
+        self.found(raise_error=True)
+
+        import adcc
+        import psi4
+
+        mol = input_model.molecule
+        model = input_model.model
+        conv_tol = input_model.keywords.get("conv_tol", 1e-6)
+
+        psi4_molecule = psi4.core.Molecule.from_schema(dict(mol.dict(), fix_symmetry="c1"))
+        psi4.core.clean()
+        psi4.core.be_quiet()
+        psi4.set_options(
+            {
+                "basis": model.basis,
+                "scf_type": "pk",
+                "e_convergence": conv_tol / 100,
+                "d_convergence": conv_tol / 10,
+                # 'maxiter': max_iter,
+                "reference": "RHF" if mol.molecular_multiplicity == 1 else "UHF",
+            }
+        )
+        _, wfn = psi4.energy("HF", return_wfn=True, molecule=psi4_molecule)
+        adcc.set_n_threads(config.ncores)
+        compute_success = False
+        try:
+            adcc_state = adcc.run_adc(wfn, method=model.method, **input_model.keywords)
+            compute_success = adcc_state.converged
+        except adcc.InputError as e:
+            raise InputError(str(e))
+        except Exception as e:
+            raise UnknownError(str(e))
+
+        input_data = input_model.dict(encoding="json")
+        output_data = input_data.copy()
+        output_data["success"] = compute_success
+
+        if compute_success:
+            output_data["properties"] = AtomicResultProperties()
+            output_data["return_result"] = adcc_state.excitation_energy
```

Comment:
this is returning a list or array right now? I think that's working b/c array is allowed for gradient, but pretty much QCA is expecting driver energy to have a float in return_result, gradient to have gradient array, etc. The driver `properties` can return a dict. Would it be reasonable in the absence of a target state to put `adcc_state.excitation_energy[0]` in this field? I assume the full excitation_energy list is in qcvars somewhere.

Reply:
Yes, the full `excitation_energy` list is in qcvars. For me it would be okay to just return the lowest excitation energy. Is there any "precedence" for this with other excited state methods?

Reply:
We could also return the MP2 energy. I think I did that in Psi4 for the "energy" version of ADC.

Reply:
Done. Returning MP ground state energy now.